addons/zylann.hterrain/native/src/quad_tree_lod.h

```diff
@@ -0,0 +1,127 @@
+#ifndef QUAD_TREE_LOD_H
+#define QUAD_TREE_LOD_H
+
+#include <Godot.hpp>
+#include <FuncRef.hpp>
+#include <CanvasItem.hpp>
+
+#include <vector>
+
+
+namespace godot {
+
+class QuadTreeLod : public Reference {
+    GODOT_CLASS(QuadTreeLod, Reference)
+
+    static const unsigned int NO_CHILDREN = -1;
+    static const unsigned int ROOT = -1;
+
+    class Quad {
+
+    private:
+        Variant _data;          // Type is Object: HTerrainChunk.gd
+
+    public:
+        unsigned int first_child = NO_CHILDREN;
+        int origin_x = 0;
+        int origin_y = 0;
+
+        Quad() {
+            init();
+        }
+
+        ~Quad() {
+        }
+
+        inline void init() {
+            first_child = NO_CHILDREN;
+            origin_x = 0;
+            origin_y = 0;
+            clear_data();
+        }
+
+        inline void clear_data() {
+            _data = Variant();
+        }
+
+        inline bool has_children() {
+            return first_child != NO_CHILDREN;
+        }
+
+        inline bool is_null() {
+            return _data.get_type() == Variant::NIL;
+        }
+
+        inline bool is_valid() {
+            return _data.get_type() != Variant::NIL;
+        }
+
+        inline Variant get_data() {
+            return _data;
+        }
+
+        inline void set_data(Variant p_data) {
+            _data = p_data;
+        }
+
+    };
+
+private:
+    Quad _root;
+    std::vector<Quad> _tree;
```

Comment:
This is technically not a tree but a pool of quads, not sure if the name is accurate

Reply:
renamed